core/trino-main/src/main/java/io/trino/operator/PagesIndex.java

```diff
@@ -521,6 +521,7 @@ public LookupSourceSupplier createLookupSourceSupplier(
                     valueAddresses,
```

Comment:
nit: still, I'm curious about microbenchmarks :). I recall I did something similar once and it didn't help

Reply:
No clue. Micro and Macro benchmarks has long been independent. I'd guess that JIT is doing different magic tricks in different environments